Armadillo/src/main/java/com/scribd/armadillo/error/ArmadilloException.kt

```diff
@@ -89,61 +108,71 @@ object UnableToLoadDownloadInfo : ArmadilloException(Exception()) {
  * the background before DownloadService.sendAddDownload
  */
 
-data class DownloadServiceLaunchedInBackground(val id: Int) : ArmadilloException(Exception()) {
+data class DownloadServiceLaunchedInBackground(val id: Int, override val cause: Exception)
+    : ArmadilloException(cause = cause,
+    message = "Android 12 compliance problem: ${cause.message}") {
     override val errorCode = 304
 }
 
-data class UnexpectedDownloadException(val throwable: Throwable): ArmadilloException(exception = Exception(throwable)){
+class UnexpectedDownloadException(throwable: Throwable)
+    : ArmadilloException(cause = throwable, message = "Unknown problem while downloading."){
     override val errorCode = 305
 }
 
 /**
  * Misc Errors
  */
-data class UnexpectedException(val exception: Exception) : ArmadilloException(exception) {
-    constructor(reason: String) : this(Exception(reason))
-
+class UnexpectedException(cause: Exception, actionThatFailedMessage: String)
```

Comment:
NP: This uses `actionThatFailedMessage` compared to `whatActionFailedMessage` in `ArmadilloIOException`

Might not be a bad idea to keep them synced

Reply:
ok